demos/sample_azure_iot_pnp/sample_azure_iot_pnp_simulated_data.c

```diff
@@ -0,0 +1,497 @@
+
+    if( xResult != eAzureIoTSuccess )
+    {
+        LogError( ( "Error getting the property version: result 0x%08x", xResult ) );
+    }
+    else
+    {
+        /* Reset JSON reader to the beginning */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        while( ( xResult = AzureIoTHubClientProperties_GetNextComponentProperty( &xAzureIoTHubClient, &xReader,
+                                                                                 pxMessage->xMessageType, xPropertyType,
+                                                                                 &pucComponentName, &ulComponentNameLength ) ) == eAzureIoTSuccess )
+        {
+            if( ulComponentNameLength > 0 )
+            {
+                LogInfo( ( "Unknown component name received" ) );
+
+                /* Unknown component name arrived (there are none for this device).
+                 * We have to skip over the property and value to continue iterating */
+                prvSkipPropertyAndValue( &xReader );
+            }
+            else if( AzureIoTJSONReader_TokenIsTextEqual( &xReader,
+                                                          ( const uint8_t * ) sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT,
+                                                          sizeof( sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT ) - 1 ) )
+            {
+                xResult = AzureIoTJSONReader_NextToken( &xReader );
+                configASSERT( xResult == eAzureIoTSuccess );
+
+                /* Get desired temperature */
+                xResult = AzureIoTJSONReader_GetTokenDouble( &xReader, pxOutTemperature );
+
+                if( xResult != eAzureIoTSuccess )
+                {
+                    LogError( ( "Error getting the property version: result 0x%08x", xResult ) );
+                    break;
+                }
+
+                xResult = AzureIoTJSONReader_NextToken( &xReader );
+                configASSERT( xResult == eAzureIoTSuccess );
+            }
+            else
+            {
+                LogInfo( ( "Unknown property arrived: skipping over it." ) );
+
+                /* Unknown property arrived. We have to skip over the property and value to continue iterating. */
+                prvSkipPropertyAndValue( &xReader );
+            }
+        }
+
+        if( xResult != eAzureIoTErrorEndOfProperties )
+        {
+            LogError( ( "There was an error parsing the properties: result 0x%08x", xResult ) );
+        }
+        else
+        {
+            LogInfo( ( "Successfully parsed properties" ) );
+            xResult = eAzureIoTSuccess;
+        }
+    }
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Property message callback handler
+ */
+void vHandleProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
```

Comment:
>vHandleProperties [](http://example.com/codeflow?start=5&length=17)

add write property in the name

Reply:
live vHandleWritableProperties ? Makes sense. Done.